### DISCUSSION TITLE
How to render two flows in one component?

### DISCUSSION THREAD

**Discussion by trinhtrungit** [xyflow/xyflow#2155] [Q&A] [answered]
I have a question, I try to render 2 flows by using 2 tags **ReactFlow** with input data are **element1** and **element2** but it works with 1 in 2 input data, please help me in this case to render 2 flows. Thanks!

> return ( `<div>`
                             `<ReactFlow`
                              elements={**element1**}
                              style={{ height:  "3000px", width: "400 px", position:"absolute"}}
                              onlyRenderVisibleElements={true}
                              edgeTypes={edgeTypes}
                              key="edges"
                          >
                          `</ReactFlow>`
                          `<ReactFlow`
                              elements={**element2**}
                              style={{ height: "3000px", width: "2000 px", left: 400, position:"absolute"}}
                              onlyRenderVisibleElements={true}
                              nodeTypes={nodeTypes}
                              edgeTypes={edgeTypes}
                              key="edges"
                          >`</ReactFlow>`
                    `</div`
             )

**Comment by moklick** [comment, marked as answer]
You need to wrap each flow with a `<ReactFlowProvider />` so that they have separate stores.

**Reply by trinhtrungit** [reply]
Many thanks @moklick, it works fine to me!
